### PR TITLE
Sparta 457 checkpoint hdfs

### PR DESCRIPTION
**Pull request by dvallejo**
**Given** a fresh installation of Sparta
**When** I edit the conf file /etc/sds/sparta/application.conf
**Then** there are a property indicating the base path for checkpointing in local filesystem with value /var/sds/sparta/checkpoint

**Given** a fresh installation of Sparta
**When** I edit the conf file /etc/sds/sparta/application.conf
**Then** there are a property indicating the base path for checkpointing in HDFS with value /user/stratio/sparta/checkpoint

**Given** a policy created without checkpoint path configured
**When** I run the policy in local
**Then** it should persist the checkpoint at local_base_path_value/policy_name

**Given** a policy created without checkpoint path configured
**When** I run the policy in local
**Then** it should persist the checkpoint in HDFS at hdfs_base_path_value/policy_name

**Given** a policy created with checkpoint path value hdfs://user/stratio/another_folder
 **When** I run the policy in local
**Then** it should persist the checkpoint in HDFS at hdfs://user/stratio/another_folder/policy_name

### CODE REVIEW

File: dist/src/main/resources/application.conf [hunk 1]
```diff
@@ -16,6 +16,8 @@ sparta {
     executionMode = local
     rememberPartitioner = true
     stopGracefully = true
+    checkpointPath = "/var/log/sds" #Default in local mode
```
Comment (danielcsant) [inline dist/src/main/resources/application.conf]: default checkpoint path should be under `/var/log/sds/sparta`

File: driver/src/main/scala/com/stratio/sparta/driver/util/HdfsUtils.scala [hunk 2]
```diff
@@ -55,8 +55,18 @@ object HdfsUtils extends SLF4JLogging {
...
   private final val HdfsDefaultPort = 8020
 
-  def apply(user: String, master: String, port: Int = HdfsDefaultPort): HdfsUtils = {
-    val conf = new Configuration()
+  def hdfsConfiguration(configOpt: Option[Config]): Configuration =
+    configOpt.map { config =>
+      val master = config.getString("hdfsMaster")
+      val port = Try(config.getInt("hdfsPort")).getOrElse(HdfsDefaultPort)
+      val conf = new Configuration()
+      conf.set(DefaultFSProperty, s"hdfs://$master:$port/user/stratio")
```
Comment (danielcsant) [inline driver/src/main/scala/com/stratio/sparta/driver/util/HdfsUtils.scala]: this property should be s"hdfs://$master:$port/user/stratio/sparta" right?

File: driver/src/test/resources/reference.conf [hunk 3]
```diff
@@ -22,6 +22,7 @@ sparta {
     executionMode = local
     rememberPartitioner = true
     topGracefully = true
+    checkpointPath = "/var/log/sds"
```
Comment (danielcsant) [inline driver/src/test/resources/reference.conf]: "/var/log/sds/sparta"

File: serving-api/src/main/resources/reference.conf [hunk 4]
```diff
@@ -17,6 +17,8 @@ sparta {
     rememberPartitioner = true
     stopGracefully = true
     stopTimeout = 30000
+    #checkpointPath = "/user/stratio/sparta"  //Default value in cluster mode
+    checkpointPath = "/var/log/sds" //Default value in local mode
```
Comment (danielcsant) [inline serving-api/src/main/resources/reference.conf]: "/var/log/sds/sparta"

**Comment by danielcsant** [comment]
Please delete all checkpoint properties under examples folder

**Comment by dvallejo** [comment]
@danielcsant  Done! Thanks for the review

**Comment by danielcsant** [comment]
There are checkpoint references under examples folder

**Comment by danielcsant** [comment]
+1

**Comment by gbecares** [comment]
**Given** a fresh installation of Sparta
**When** I edit the conf file /etc/sds/sparta/application.conf
**Then** there are a property indicating the base path for checkpointing in local filesystem with value /var/log/sds/sparta/checkpoint

_Value in that file is checkpointPath = "/var/log/sds/sparta" #Default in local mode_

**Given** a fresh installation of Sparta
**When** I edit the conf file /etc/sds/sparta/application.conf
**Then** there are a property indicating the base path for checkpointing in HDFS with value /user/stratio/sparta/checkpoint

_# checkpointPath = "/user/stratio/sparta" #Default in cluster mode_

Is it correct?

**Comment by danielcsant** [comment]
@gbecares it is ok. We can renamed later if we use another functionality than the checkpointing

**Comment by danielcsant** [comment]
Running os Mesos gives the following error:

16/06/28 16:19:29 ERROR SpartaJob$: Permission denied: user=stratio, access=WRITE, inode="/var/log/sds/sparta/asdsad":hdfs:supergroup:drwxr-xr-x
	at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.check(FSPermissionChecker.java:319)
	at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.check(FSPermissionChecker.java:292)
	at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkPermission(FSPermissionChecker.java:213)
	at org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.checkPermission(FSPermissionChecker.java:190)
	at org.apache.hadoop.hdfs.server.namenode.FSDirectory.checkPermission(FSDirectory.java:1720)
	at org.apache.hadoop.hdfs.server.namenode.FSDirectory.checkPermission(FSDirectory.java:1704)
	at org.apache.hadoop.hdfs.server.namenode.FSDirectory.checkAncestorAccess(FSDirectory.java:1687)
	at org.apache.hadoop.hdfs.server.namenode.FSDirMkdirOp.mkdirs(FSDirMkdirOp.java:71)
	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.mkdirs(FSNamesystem.java:3894)
	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.mkdirs(NameNodeRpcServer.java:983)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.mkdirs(ClientNamenodeProtocolServerSideTranslatorPB.java:622)
	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:616)
	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:969)
	at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2049)
	at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2045)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
	at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2043)

**Comment by gbecares** [comment]
:shipit:

**Comment by stratiocommit** [comment]
Comment accepted. Will treat it ASAP on [Jenkins](http://jenkins.stratio.com/job/Sparta/job/CD/3035/)

**Comment by stratiocommit** [comment]
Rebase && testing in progress

**Pull request merged**